### DISCUSSION TITLE
What's the next step

### DISCUSSION THREAD

**Discussion by yeaharh** [microsoft/typespec#2675] [General]
I've create an OpenAPI specification using Typespec. Now what to do next? I'm following a an API First Design methodology. I'm thought I could find tools to autogenerate server stubs and client calling applications in dotnetcore, but this is not the case.  What tools can I expect to successfully generating server and client code. I would be happy if I could generate just the class models a client or server implementation would use.

**Comment by markcowl** [comment]
Excellent question!  There will soon be a c-sharp service emitter (there is a WIP PR for this here: https://github.com/microsoft/typespec/pull/2458), and additional code emitters in other languages in the coming months.  Meanwhile, you can find emitters for [protobuf](https://microsoft.github.io/typespec/standard-library/protobuf/reference), [json-schema](https://microsoft.github.io/typespec/standard-library/json-schema/reference), and a [sample emitter for typescript](https://github.com/microsoft/typespec/blob/main/packages/compiler/test/emitter-framework/typescript-emitter.ts).

The [emitter-framework](https://microsoft.github.io/typespec/extending-typespec/emitter-framework) provides a simpler experience for creating emitters for various kinds of artifacts, including client and service code:
